### PR TITLE
Emphasize build result in console

### LINKED ISSUE #1982: Build result should be easy to see relative to other logging
BUILD SUCCESSFUL or FAILED is the most important piece of output, and yet it is not highlighted in any way other than color sometimes.

### Expected Behavior
Let's make that bolded at least, if not add color in more cases

### Current Behavior
screen shot 2017-05-09 at 11 37 38 am

### PR DESCRIPTION
**Pull request by w25r**
### Context
Fixes #1982

### Contributor Checklist
- [x] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [x] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [x] Ensure that tests pass locally: `./gradlew quickCheck <impacted-subproject>:check`

### Gradle Core Team Checklist
- [x] Verify design and implementation 
- [x] Verify test coverage and [CI build status](https://builds.gradle.org/project.html?projectId=Gradle_Branches&branch_Gradle_Branches=wl-emphasize-build-status)
- [x] Verify documentation including proper use of `@since` and `@Incubating` annotations for all public APIs

### CODE REVIEW

**Comment by w25r** [comment]
### Normal Console
screen shot 2017-05-10 at 11 31 54 pm

### Plain Console
screen shot 2017-05-10 at 11 35 04 pm

I added a new line before and after the build message.  The `<=============>` matches what the progress bar looks like when the build completes, so it has some nice symmetry.  But it also looks okay in the plain console (no progress bar).  I made the `BUILD SUCCESSFUL` message both bold, and green.  

I expect some debate on this.  Thoughts @adammurdoch, @eriwen, @big-guy and anyone else?

**Comment by eriwen** [comment]
@w25r what is the reasoning behind keeping a full progress bar around? Suppose the build failed halfway through.

**Comment by w25r** [comment]
It isn't a `progress bar` as much as just an indicator... I experimented with `---------------------` and `======================` and I thought the best user experience was the length matching, rather than being longer.  And since `-----` becomes `=====` it led me to that.  

I should add the failure image too.  I will do that now.

**Comment by eriwen** [comment]
I'm still not sure what the bar is an indicator of. I think what we had is sufficient and we just need a little styling tbh

**Comment by bmuschko** [comment]
I think users might confuse `<=============>` or `=============` with the progress bar. It simply looks too similar. We might be able to think about other ways to separate the build result based on feedback from end user. For now I'd just go with bold and green and push this PR through.

**Comment by w25r** [comment]
I've now bolded both `BUILD FAILED`and `BUILD SUCCESSFUL` messages.  

I also needed to do some refactoring to maintain passivity.  If the property `org.gradle.color.failure` is set to a color other than red, we need to continue to respect that setting.

File: subprojects/logging/src/main/java/org/gradle/internal/logging/console/DefaultColorMap.java [hunk 1]
```diff
@@ -99,33 +125,60 @@ public Color getColourFor(Style style) {
...
-            colors.put(style, color);
+            colorByStyle.put(style, color);
         }
 
         return color;
     }
 
+    private String getColorSpecForStyle(StyledTextOutput.Style style) {
+        String lowerCase = style.name().toLowerCase();
+        return System.getProperty("org.gradle.color." + lowerCase, defaults.get(lowerCase));
```
Comment (lacasseio) [inline subprojects/logging/src/main/java/org/gradle/internal/logging/console/DefaultColorMap.java]: :o: Can we unify this property name with the one below (line 143)?

File: subprojects/logging/src/test/groovy/org/gradle/internal/logging/console/DefaultColorMapTest.groovy [hunk 2]
```diff
@@ -47,6 +47,56 @@ class DefaultColorMapTest extends Specification {
...
+        given:
+        System.properties['org.gradle.color.successheader'] = 'italic-cyan'
+        System.properties['org.gradle.color.failure'] = 'magenta'
+        System.properties['org.gradle.color.header'] = 'italic'
+        System.properties['org.gradle.color.error'] = 'bold-red'
+        //For this test the color map needs to be created after the system properties are set
+        this.map = new DefaultColorMap()
+        Ansi ansi = Mock()
+
+        when: 'A color is requested for SuccessHeder which is overridden'
```
Comment (lacasseio) [inline subprojects/logging/src/test/groovy/org/gradle/internal/logging/console/DefaultColorMapTest.groovy]: Nice use of the comment for the behavior tags!

**Pull request merged**